tools/dailyjobs_query.py

```diff
@@ -0,0 +1,126 @@
+                        help="starting date."
+                        )
+
+    parser.add_argument("-e", "--end_date",
+                        metavar="YYYY-MM-DD",
+                        default = datetime.date.today(),
+                        dest="end_date",
+                        help="ending date."
+                        )
+
+    options = parser.parse_args(argv)
+    return options
+
+
+def updatedb(date, platform, branch, numpushes, numjobs, sumduration):
+    cur.execute('delete from dailyjobs where date="%s" and branch="%s" and platform="%s"' % (date, branch, platform))
+    query = 'insert into dailyjobs (date, platform, branch, numpushes, numjobs, sumduration) values ("{0}", "{1}", "{2}", {3}, {4}, {5})'
+    query = query.format(date, platform, branch, numpushes, numjobs, sumduration)
+    print query
+    cur.execute(query)
+
+
+def summarize(date, branch):
+    for platform in platforms:
+        numpushes = 0
+        numjobs = 0
+        sumduration = 0
+        for rev in revisions_dict:
+            if platform not in revisions_dict[rev]:
+                continue
+
+            value = revisions_dict[rev][platform]
+            if value[0] != date:
+                continue
+
+            numpushes += 1
+            numjobs += value[1]
+            sumduration += value[2]
+        print (date, platform, branch, numpushes, numjobs, sumduration)
+        updatedb(date, platform, branch, numpushes, numjobs, sumduration)
+
+
+def retrievedb(branch, date):
+    query = "select revision,count(result),sum(duration),platform from testjobs where branch='{0}' and date like '{1}%' group by revision,platform;"
+    query = query.format(branch, date)
+    print query
+    cur.execute(query)
+
+    for rows in cur.fetchall():
+        print "%s, %s" % (date, rows)
+        revision = rows[0]
+        jobs = int(rows[1])
+        duration = int(rows[2])
+        platform = rows[3]
+        found = False
+        for p in platforms:
+            if platform.startswith(p):
+                platform = p
+                found = True
+                break
+
+        # we can skip b2g/mulet or other platforms that don't match our core set
+        if not found:
+            continue
+
+        if revision not in revisions_dict:
+            revisions_dict[revision] = {}
+            for p in platforms:
+                revisions_dict[revision][p] = [date, 0, 0]
+
+        revisions_dict[revision][platform][1] += jobs
+        revisions_dict[revision][platform][2] += duration
+
+
+if __name__ == "__main__":
+    options = parse_args()
+
+    if type(options.start_date) == str:
+        options.start_date = datetime.datetime.strptime(options.start_date, "%Y-%m-%d").date()
+    if type(options.end_date) == str:
+        options.end_date = datetime.datetime.strptime(options.end_date, "%Y-%m-%d").date()
+
+    if options.start_date and not options.end_date:
+        options.end_date = options.start_date
```

Comment:
Another thing you could check is whether the end_date is before the start_date and either issue an error or swap the dates.